matrix_client/api.py

```diff
@@ -97,20 +97,15 @@ def validate_certificate(self, valid):
         self.validate_cert = valid
         return
 
-    def register(self, login_type, **kwargs):
+    def register(self, content={}, query_params={}):
         """Performs /register.
 
         Args:
-            login_type(str): The value for the 'type' key.
-            **kwargs: Additional key/values to add to the JSON submitted.
+            content(dict): The request payload. Should include "type" such as "m.login.password" for all non-guest registrations.
+            query_params(dict): The query params for the request. Specify "kind": "guest" to register a guest account.
```

Comment:
I'd prefer to have a single parameter called kind for now rather than an open ended ``query_param`` option. I realise you are trying to future proof the API, but in this case I'd prefer clarity and slightly more readable functions. Content is fine in this regard, because it tends to have multiple sets of keys.

Oh and make it clear that it defaults to user, just to make it totally obvious.